### PR TITLE
[Release 1.27] Add e2e test about dnscache

### PR DESCRIPTION
**Pull request by manuelbuil**
Backport: https://github.com/rancher/rke2/commit/f8605328d2cb2c7768518900ed04d3170d0b8ddf
Issue: https://github.com/rancher/rke2/issues/5221

### CODE REVIEW

**Review by dereknola — APPROVED** [review]

File: tests/e2e/dnscache/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,99 @@
+ENV['VAGRANT_NO_PARALLEL'] = 'no'
+NODE_ROLES = (ENV['E2E_NODE_ROLES'] || ["server-0", "agent-0" ])
+NODE_BOXES = (ENV['E2E_NODE_BOXES'] || ['generic/ubuntu2004', 'generic/ubuntu2004'])
+GITHUB_BRANCH = (ENV['E2E_GITHUB_BRANCH'] || "master")
```
Comment (dereknola) [inline tests/e2e/dnscache/Vagrantfile:4]:
```suggestion
GITHUB_BRANCH = (ENV['E2E_GITHUB_BRANCH'] || "release-1.27")
```

File: tests/e2e/dnscache/dnscache_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+)
+
+// Valid nodeOS: generic/ubuntu2004, opensuse/Leap-15.3.x86_64
+var nodeOS = flag.String("nodeOS", "generic/ubuntu2004", "VM operating system")
+var serverCount = flag.Int("serverCount", 1, "number of server nodes")
+var agentCount = flag.Int("agentCount", 1, "number of agent nodes")
+var ci = flag.Bool("ci", false, "running on CI")
+
+// Environment Variables Info:
+// E2E_RELEASE_VERSION=v1.23.1+rke2r1 or nil for latest commit from master
```
Comment (dereknola) [inline tests/e2e/dnscache/dnscache_test.go:23]:
```suggestion
// E2E_RELEASE_VERSION=v1.23.1+rke2r1 or nil for latest commit from release-1.27
```

**Review by dereknola — CHANGES_REQUESTED** [review]

File: tests/e2e/dnscache/Vagrantfile (hunk 1 above)
Comment (manuelbuil) [reply tests/e2e/dnscache/Vagrantfile]: thanks!

**Review by dereknola — APPROVED** [review]
Please revert the changes based on my previous comments.

**Review by brandond — APPROVED** [review]

**Pull request merged**